open_fdd/desktop/ui/main_window.py

```
@@ -605,31 +726,82 @@ def _on_run_rules(self) -> None:
+            end_dt = QDateTime.fromString(str(info["end"]), Qt.ISODate)
+            if start_dt.isValid():
+                self.rules_start_dt.setDateTime(start_dt)
+            if end_dt.isValid():
+                self.rules_end_dt.setDateTime(end_dt)
+            self.rules_use_window.setChecked(True)
+            self.rules_out.setPlainText(
+                f"Loaded bounds for source '{source}' ({info['rows']} rows)\n"
+                f"Start: {self.rules_start_dt.dateTime().toString(Qt.ISODate)}\n"
+                f"End: {self.rules_end_dt.dateTime().toString(Qt.ISODate)}"
+            )
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)